packages/theme-language-server-common/src/server/startServer.ts

```diff
@@ -171,6 +173,23 @@ export function startServer(
     return getDefaultSchemaTranslations();
   };
 
+  const getLiquidDocDefinitionsForURI = async (
```

Comment:
I want to memoize this in a follow-up PR, however we would need to invalidate the cache if the file gets updated or deleted.

I see that we do something kind of similar for metafields [here](https://github.com/Shopify/theme-tools/blob/3b20751ddae8466e59e84f1fbfa0c631a981e6cc/packages/theme-language-server-common/src/server/startServer.ts#L510-L513)

I would want to add this logic in a follow-up outside of this PR, anyhow.

@aswamy before I dive into this, is the `memoize` function a good fit here?